snapcraft/internal/meta.py

```diff
@@ -163,25 +164,28 @@ def write_snap_directory(self):
                     os.remove(destination)
                 file_utils.link_or_copy(source, destination)
 
-        # Now copy the hooks contained within the snap directory directly into
-        # meta (they don't get wrappers like the ones that come from parts).
-        snap_hooks_dir = os.path.join('snap', 'hooks')
-        hooks_dir = os.path.join(self._snap_dir, 'meta', 'hooks')
-        if os.path.isdir(snap_hooks_dir):
-            os.makedirs(hooks_dir, exist_ok=True)
-            for hook_name in os.listdir(snap_hooks_dir):
-                source = os.path.join(snap_hooks_dir, hook_name)
-                destination = os.path.join(hooks_dir, hook_name)
-
-                # First, verify that the hook is actually executable
-                if not os.stat(source).st_mode & stat.S_IEXEC:
-                    raise CommandError('hook {!r} is not executable'.format(
-                        hook_name))
-
-                with contextlib.suppress(FileNotFoundError):
-                    os.remove(destination)
-
-                file_utils.link_or_copy(source, destination)
+        # Now copy the assets contained within the snap directory directly into
+        # meta (they don't get wrappers for hooks like the ones that come from
+        # parts).
+        for origin in ['gui', 'hooks']:
+            src_dir = os.path.join('snap', origin)
+            dst_dir = os.path.join(self.meta_dir, origin)
+            if os.path.isdir(src_dir):
+                os.makedirs(dst_dir, exist_ok=True)
+                for asset in os.listdir(src_dir):
+                    source = os.path.join(src_dir, asset)
+                    destination = os.path.join(dst_dir, asset)
+
+                    # First, verify that the hook is actually executable
+                    if (origin == 'hooks' and
+                            not os.stat(source).st_mode & stat.S_IEXEC):
+                        raise CommandError(
+                            'hook {!r} is not executable'.format(asset))
```

Comment:
Would it be better/more clear to refactor this into a standalone function that accepts a path and some sort of `checker` function?